util.go

```diff
@@ -0,0 +1,59 @@
+package jwt
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+// ByteArrayVal interface to byte array
+func ByteArrayVal(value interface{}) (key []byte) {
+	if value == nil {
+		return
+	}
+	switch value.(type) {
```

Comment:
Instead of doing the CAST within each SWITCH CASE you can create a variable already typed.


Example:
```golang
switch typedValue := value.(type) {
....
}
```

Official documentation: https://go.dev/tour/methods/16